plugins/module_utils/acme/challenges.py

```diff
@@ -47,6 +47,11 @@ def combine_identifier(identifier_type, identifier):
     return '{type}:{identifier}'.format(type=identifier_type, identifier=identifier)
 
 
+def normalize_combined_identifier(identifier):
+    parts = split_identifier(identifier)
+    return '{type}:{identifier}'.format(type=parts[0], identifier=parts[1].lower())
```

Comment:
I would split this up a bit more (and also use `combine_identifier`):
```suggestion
    identifier_type, identifier = split_identifier(identifier)
    # Normalize DNS names and IPs
    identifier = identifier.lower()
    return combine_identifier(identifier_type, identifier)
```